qiskit_dynamics/models/operator_collections.py

```diff
@@ -33,104 +33,126 @@
-    given the 1d set of values :math:`c_j`.
+    Describes an interface for evaluating the map or its action on ``y``, given the 1d set of values
+    :math:`c_j`.
     """
 
-    def __init__(
-        self,
-        static_operator: Optional[any] = None,
-        operators: Optional[any] = None,
-    ):
+    def __init__(self, static_operator: Optional[Any] = None, operators: Optional[Any] = None):
         """Initialize.
 
-        Accepted types are determined by concrete subclasses.
-
         Args:
-            operators: (k,n,n) Array specifying the terms :math:`G_j`.
-            static_operator: (n,n) Array specifying the extra static_operator :math:`G_d`.
+            operators: ``(k,n,n)`` array specifying the terms :math:`G_j`.
+            static_operator: ``(n,n)`` array specifying the extra static_operator :math:`G_d`.
         """
         self.operators = operators
         self.static_operator = static_operator
 
+    @property
+    def dim(self) -> int:
+        """The matrix dimension."""
+        if self.static_operator is not None:
+            return self.static_operator.shape[-1]
+        else:
+            return self.operators[0].shape[-1]
+
     @property
     def static_operator(self) -> Array:
-        """Returns static part of operator collection."""
+        """The static part of the operator collection."""
 
     @static_operator.setter
     def static_operator(self, new_static_operator: Optional[Array] = None):
-        """Sets static_operator term."""
+        pass
 
     @property
     def operators(self) -> Array:
-        """Return operators."""
+        """The operators of this collection."""
 
     @operators.setter
     def operators(self, new_operators: Array) -> Array:
-        """Return operators."""
+        pass
 
     @abstractmethod
     def evaluate(self, signal_values: Array) -> Array:
-        r"""Evaluate the map."""
+        r"""Evaluate the operator :math:`\Lambda(c, \cdot) = (G_d + \sum_jc_jG_j)`.
+
+        Args:
+            signal_values: The signals values :math:`c` to use on the operators.
+
+        Returns:
+            An :class:`~Array` that acts on states ``y`` via multiplication.
+        """
 
     @abstractmethod
     def evaluate_rhs(self, signal_values: Union[List[Array], Array], y: Array) -> Array:
-        r"""Compute the function."""
+        r"""Evaluate the function and return :math:`\Lambda(c, y) = (G_d + \sum_jc_jG_j)  y`.
+
+        Args:
+            signal_values: The signals :math:`c` to use on the operators.
+            y: The system state.
+
+        Returns:
+            The evaluated function.
+        """
 
     def __call__(
         self, signal_values: Union[List[Array], Array], y: Optional[Array] = None
     ) -> Array:
-        """Call either ``self.evaluate`` or ``self.evaluate_rhs`` depending on number of
-        arguments.
-        """
+        """Call :meth:`~evaluate` or :meth:`self.evaluate_rhs` depending on the presense of ``y``.
```

Comment:
What's the difference between using the tilde and directly referencing `self.`?

Reply:
The discrepancy here is indeed unwarranted, will fix. (tilde suppresses ptah of method:`qiskit_dynamics.models.OperatorCollection.evaluate` vs just `evaluate`)